src/main/java/com/google/impactdashboard/server/api_utilities/ResourceRetriever.java

```diff
@@ -75,6 +77,60 @@ protected ProjectListRetriever(CloudResourceManager cloudResourceManagerService)
     return projects;
   }
 
+  /**
+   * Returns the top level ancestor to a project which is the organization the project belongs under.
+   * @param projectId the project the organization id is being retrieved for.
+   */
+  public String getOrganizationId(String projectId){
+    List<Ancestor> ancestors;
+    try {
+       ancestors = getProjectAncestry(projectId);
+    } catch (IOException io) {
+      throw new RuntimeException("Failed to get ancestors: " + io.getMessage());
+    }
+    if(ancestors == null) {
```

Comment:
if this is a magic string that must be interpreted elsewhere, please make it a static function.

Reply:
If you are talking about the return of this function it is not magic it is just the organization id which is stored in the database as a string